### DISCUSSION TITLE
[BUG] Regex Policy based on multivalued attributes

### DISCUSSION THREAD

**Discussion by immieszko** [sventorben/keycloak-restrict-client-auth#266] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Using [keycloak-restrict-client-auth](https://github.com/sventorben/keycloak-restrict-client-auth)` with Policy-Based mode. When I want to use Regex Client Policy on multivalue attributes it does not work. 
For example, I have User user1 with an attribute:
```
"attr": [
    "1test",
    "2test"
  ],
```
![image](https://github.com/sventorben/keycloak-restrict-client-auth/assets/56539674/f85d8a83-0824-45f6-a1cb-42f03cc96cec)

If I define the target_claim to my_attr and regex pattern to 1test it works, but when I would like to check the 2test I have no idea how to do it :disappointed:. Do you know any solution?
![image](https://github.com/sventorben/keycloak-restrict-client-auth/assets/56539674/201db826-cbe1-494a-84af-2bccbf98a7de)

### Expected Behavior

Check each value from multivalued attribute not only first.

### Steps To Reproduce

1. Create Policy-Based flow,
2. Use Regex policy based on multivalued attribute
3. Check the second value of attribute in Policy

### Version

```markdown
- Keycloak: 24.0.1
- This extension: 24.0.0
```

**Comment by sventorben** [comment, marked as answer]
Hey @immieszko

I moved this to the discussion section since this is not a bug within my extension. What you are using there is a default Keycloak feature.
Anyways, I don't think that what you are trying to achieve will work, because the the RegexPolicyProvider will only use the first attribute value. See https://github.com/keycloak/keycloak/blob/24.0.1/authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/regex/RegexPolicyProvider.java#L99

I'd recommend you create a feature request at the Keycloak project to support multi-valued attributes here.
